Source/Editor/Windows/ContentWindow.ContextMenu.cs

```diff
@@ -159,7 +189,17 @@ private void ShowContextMenuForItem(ContentItem item, ref Float2 location, bool
                 var p = Editor.ContentDatabase.Proxy[i];
                 if (p.CanCreate(folder))
                 {
-                    c.ContextMenu.AddButton(p.Name, () => NewItem(p));
+                    if (_animationGroupNames.Contains(p.Name))
```

Comment:
I think it would be better to let `AssetProxy` return the name of the category and use it here to build nested menu - easier to extend and maintain in the future.

Reply:
I am thinking of doing it this way. Each asset proxy will have its category name that is set. Then use something like this:

```csharp
 // categorize the asset proxies according to their category name
            Dictionary<string, List<AssetProxy>> categorizedProxyList = new Dictionary<string, List<AssetProxy>>();
            for (int i = 0; i < Editor.ContentDatabase.Proxy.Count; i++)
            {
                var p = Editor.ContentDatabase.Proxy[i];
                if (p.CanCreate(folder))
                {
                    if (p is AssetProxy ap)
                    {
                        if (categorizedProxyList.ContainsKey(ap.CategoryName))
                        {
                            categorizedProxyList.TryGetValue(ap.CategoryName, out var apList);
                            apList?.Add(ap);
                        }
                        else
                        {
                            categorizedProxyList.Add(ap.CategoryName, new List<AssetProxy>() {ap});
                        }
                    }
                }
            }
            
            c = cm.AddChildMenu("New");
            c.ContextMenu.Tag = item;
            c.ContextMenu.AutoSort = true;
            int newItems = 0;
            for (int i = 0; i < Editor.ContentDatabase.Proxy.Count; i++)
            {
                var p = Editor.ContentDatabase.Proxy[i];
                if (p.CanCreate(folder))
                {
                    if (p is AssetProxy ap && categorizedProxyList.TryGetValue(ap.CategoryName, out var apList))
                    {
                        if (apList.Contains(ap))
                        {
                            if (apList.Count > 1)
                            {
                                var childMenu = c.ContextMenu.GetOrAddChildMenu(ap.CategoryName);
                                childMenu.ContextMenu.AutoSort = true;
                                childMenu.ContextMenu.AddButton(p.Name, () => NewItem(p));
                            }
                            else
                            {
                                c.ContextMenu.AddButton(p.Name, () => NewItem(p));
                            }
                        }
                    }
                    else
                    {
                        c.ContextMenu.AddButton(p.Name, () => NewItem(p));
                    }

                    newItems++;
                }
            }
    ```

Can you think of an easier way besides using 2 loops?